### ISSUE TITLE
Measure whether dexter's NPU buys whisper.cpp a bigger model at the same latency

### ISSUE THREAD

**Issue by hf7y** [hf7y/crt#123]
NO-DECISION: Zach's own drop, verbatim below, recovered from the 2026-08-30 idea residue and filed where the STT actually runs

Dropped 2026-07-22 and never filed anywhere. It came up while discussing this
repo's whisper.cpp setup, so it is filed here rather than against the host.

> clean up dexter (the Ryzen mini-PC running crt-vm) and wire in its real NPU
> (Ryzen AI/XDNA) tooling. Came up while discussing crt's whisper.cpp STT
> setup: the NPU can't do training/fine-tuning (it's quantized low-power
> inference silicon, no backprop toolchain), but it could accelerate
> whisper.cpp inference itself via the Ryzen AI SDK/DirectML/ONNX Runtime
> backend, letting the console run a bigger/more accurate model than base.en at
> the same latency. Would need a different whisper.cpp build/backend than the
> plain-CPU one install.sh builds now.

The claim to test first, before any build work: that a Ryzen AI backend gives
this console a bigger model at the same latency. That is a measurement on one
machine with one audio file, and it decides whether the rest is worth doing.

Note the host has changed under this since it was written: dexter now runs
monkey as a WSL2 distro and holds AMD-V, so anything touching its virtualisation
stack is senechal's, not this repo's. The whisper backend is not.

DEFAULT-AFTER 0d: block -- no default. A hardware-acceleration path is worth
measuring before it is chosen, and nothing breaks while it is unmeasured.

- none

- host:dexter -- a measured latency/accuracy comparison, CPU whisper.cpp vs a Ryzen AI backend

**Comment by hf7y** [comment]
NO-DECISION: triage -- labeling needs-host, the measurement genuinely needs dexter's own Windows/NPU toolchain

Tried the obvious path from this dev clone (running as \`monkey\`, the WSL2
distro this issue itself says now runs ON dexter):

    ssh -o BatchMode=yes -o ConnectTimeout=3 dexter true
    ssh: connect to host dexter port 22: Connection refused

No sshd reachable on the Windows host side, and no other access path from
here. The measurement this issue names -- a Ryzen AI/DirectML/ONNX Runtime
whisper.cpp backend vs. the plain-CPU build, one audio file, one latency
number -- needs the Ryzen AI SDK, which is Windows-host tooling, not
something reachable from inside a WSL2 guest over what's available here.
That is squarely "needs a physical device or a live remote host" per this
label's own definition, so tagging it rather than leaving a dev-only run to
re-discover the same wall next time.


- none



- none

**Comment by hf7y** [comment]
NO-DECISION: the CPU baseline this measurement needs, taken 2026-09-02, plus a correction to the reachability claim above.

**The CPU number to beat.** Measured against the live `zaxon-whisper` container on dexter -- whisper.cpp v1.9.3, stock `ggml-base.en.bin`, default `-t 4` on an 8-core host:

    11.0s of audio (relay/samples/jfk.wav)  ->  0.87s, 0.89s   (~12.6x real-time)
    3 concurrent POSTs of the same file     ->  2.64s total

The second line is the one that surprised me: `whisper-server` **serialises**. Three requests take three times one request, so any plan that hopes to buy latency by sending chunks in parallel over HTTP buys nothing. whisper.cpp's own `-p N` splits audio across processors inside one request, and `--vad -vmsd N` auto-splits long speech; both reach the server through a compose `command:` line, because `provision/dexter/whisper/entrypoint.sh` ends in `exec whisper-server ... "$@"`.

So the honest framing of this issue's claim: at 12.6x real-time on plain CPU there is **no latency problem to spend an NPU on** -- a two-minute voice note transcribes in ~10s, well inside the WhatsApp round trip. The NPU's payoff would be accuracy at constant latency (`small.en` or better in the same budget), not speed. Worth measuring for that, not for this.

**Correction to the triage comment: dexter IS reachable, on port 2223.** From mandark, `ssh dexter` works today (`~/.ssh/config` sets `Port 2223`; port 22 is refused, which is what that comment hit), and from there:

    ssh dexter "/mnt/c/Windows/System32/wsl.exe -d Ubuntu -- docker ps"

runs against the distro holding the containers. Every number above was taken that way. `wsl.exe` is not on PATH in that ssh session -- the absolute path is load-bearing.

The `needs-host` label may still be right for the *Ryzen AI SDK* half, which is Windows-side tooling. It is not right for "cannot reach the machine".

**Comment by hf7y** [comment]
DECISION (Zach, realisateur `/ideate`, 2026-09-06): **all three, in order — test accuracy first, then move to a bigger model on CPU, then close the NPU question.**

> 1,2,3

## 1. Test accuracy first

The measurement above already killed the latency argument: **12.6x real-time on plain CPU**, a two-minute voice note in ~10s, well inside a WhatsApp round trip. So the only live question is whether `base.en` is *accurate enough*, and nothing has measured that.

Two real cases, both of which now matter:

- **Voice notes.** hf7y/crt#154 was ruled today: a tagged note auto-files a GitHub issue pointing at the inbox row. A misheard tag files into the wrong repo, so transcription accuracy became load-bearing today in a way it was not last week.
- **Meeting audio.** hf7y/secretaire#38 names this exactly: *"`base.en` on far-field multi-speaker is the untested case, not the plumbing"*, and the recording that prompted the whole thread failed. This is the harder case and the one most likely to fail.

**Run both against real audio and read the output.** Not WER against a corpus — read it, and see whether it says what was said.

## 2. Then move to a bigger model on CPU

There is 12x headroom. `small.en` or `medium.en` in the same latency budget is a volume mount and a model download, not an SDK, not a new backend, not a rebuild of whisper.cpp. Do this on the strength of step 1's reading rather than pre-emptively — but the ordering means it is expected, not conditional.

Two things measured above that constrain how far the headroom goes:

- **`whisper-server` serialises.** Three concurrent POSTs took three times one POST. Parallelism over HTTP buys nothing; `-p N` and `--vad -vmsd N` are the in-request splits, reached through the compose `command:` line since `entrypoint.sh` ends in `exec whisper-server ... "$@"`.
- A bigger model eats the headroom roughly linearly. State the new real-time factor after the change so the next person does not re-derive it.

## 3. Then close the NPU question

Record that the NPU buys nothing measurable and stop tracking it. The original claim — that the Ryzen AI SDK / DirectML / ONNX Runtime backend could run a bigger model at the same latency — is not *false*, but step 2 reaches the same destination with a model download instead of a different whisper.cpp build and a Windows-side toolchain.

This issue's `needs-host` label was already noted as right only for the Ryzen AI SDK half. Closing that half removes the label's last justification.
